commands/kubernetes.go

```diff
@@ -395,22 +405,169 @@ func RunKubernetesKubeconfigShow(c *CmdConfig) error {
+func loadCachedExecCredential(id string) *clientauthentication.ExecCredential {
+	path := cachedExecCredentialPath(id)
+	f, err := os.Open(path)
+	if err != nil {
+		return nil
+	}
+
+	defer f.Close()
+
+	var execCredential *clientauthentication.ExecCredential
+	if err := json.NewDecoder(f).Decode(&execCredential); err != nil {
+		if Verbose {
+			fmt.Fprintln(os.Stderr, err)
+		}
+		return nil
+	}
+
+	if execCredential.Status == nil {
+		return nil
+	}
+
+	t := execCredential.Status.ExpirationTimestamp
+	if t.IsZero() || t.Time.Before(time.Now()) {
+		if err := os.Remove(path); err != nil && Verbose {
+			fmt.Fprintln(os.Stderr, err)
+		}
+		return nil
+	}
+
+	return execCredential
+}
+
+// cacheExecCredential attempts to cache an ExecCredential. Never errors.
+func cacheExecCredential(id string, execCredential *clientauthentication.ExecCredential) {
+	// Don't bother caching if there's no expiration set
+	if execCredential.Status.ExpirationTimestamp.IsZero() {
+		return
+	}
+
+	cachePath := kubeconfigCachePath()
+	if err := os.MkdirAll(cachePath, os.FileMode(0700)); err != nil {
+		if Verbose {
+			fmt.Fprintln(os.Stderr, err)
+		}
+		return
+	}
+
+	path := cachedExecCredentialPath(id)
+	f, err := os.OpenFile(path, os.O_CREATE|os.O_RDWR|os.O_TRUNC, os.FileMode(0600))
+	if err != nil {
+		if Verbose {
+			fmt.Fprintln(os.Stderr, err)
+		}
+		return
+	}
+	defer f.Close()
+
+	if err := json.NewEncoder(f).Encode(execCredential); err != nil && Verbose {
+		fmt.Fprintln(os.Stderr, err)
+	}
+}
+
+// RunKubernetesKubeconfigExecCredential displays the exec credential. It is for internal use only.
+func RunKubernetesKubeconfigExecCredential(c *CmdConfig) error {
+	if len(c.Args) != 1 {
+		return doctl.NewMissingArgsErr(c.NS)
+	}
+
+	version, err := c.Doit.GetString(c.NS, doctl.ArgVersion)
+	if err != nil {
+		return err
+	}
+
+	if version != "v1beta1" {
+		return fmt.Errorf("invalid version %q expected 'v1beta1'", version)
+	}
+
+	kube := c.Kubernetes()
+
+	clusterID := c.Args[0]
+	execCredential := loadCachedExecCredential(clusterID)
+	if execCredential != nil {
+		return json.NewEncoder(c.Out).Encode(execCredential)
+	}
+
+	kubeconfig, err := kube.GetKubeConfig(clusterID)
+	if err != nil {
+		return err
+	}
+
+	config, err := clientcmd.Load(kubeconfig)
+	if err != nil {
+		return err
+	}
+
+	execCredential, err = execCredentialFromConfig(config)
+	if err != nil {
+		return err
+	}
+	cacheExecCredential(clusterID, execCredential)
+
+	return json.NewEncoder(c.Out).Encode(execCredential)
+}
+
+func execCredentialFromConfig(config *clientcmdapi.Config) (*clientauthentication.ExecCredential, error) {
+	current := config.CurrentContext
+	context, ok := config.Contexts[current]
+	if !ok {
+		return nil, fmt.Errorf("received invalid config from API")
+	}
+
+	authInfo, ok := config.AuthInfos[context.AuthInfo]
+	if !ok {
+		return nil, fmt.Errorf("received invalid config from API")
```

Comment:
Error return should be fixed. `fmt.Errorf("auth info for context %q doesn't exist", contextName)`